src/type_sys/type_id.rs

```diff
@@ -26,6 +28,24 @@ impl FenderTypeId {
             String => false,
             List => false,
             Char => true,
+            Struct => false,
+            Type => true,
+        }
+    }
+    pub fn type_from_str(value: String) -> Option<FenderTypeId> {
+        match value.as_str() {
+            "Int" => Some(FenderTypeId::Int),
+            "Float" => Some(FenderTypeId::Float),
+            "Bool" => Some(FenderTypeId::Bool),
+            "Error" => Some(FenderTypeId::Error),
```

Comment:
We have so much of this kind of function relating to types now that I'm starting to think we should make a macro for it